opsdroid/loader.py

```diff
@@ -47,30 +47,35 @@ def import_module_from_spec(module_spec):
     @staticmethod
     def import_module(config):
         """Import module namespace as variable and return it."""
-        # Check if the module can be imported and proceed with import
-
-        # Proceed only if config.name is specified
-        # and parent module can be imported
-        if config["name"] and importlib.util.find_spec(config["module_path"]):
-            module_spec = importlib.util.find_spec(config["module_path"] +
-                                                   "." + config["name"])
-            if module_spec:
-                module = Loader.import_module_from_spec(module_spec)
-                _LOGGER.debug(_("Loaded %s: %s"), config["type"],
-                              config["module_path"])
-                return module
-
-        module_spec = importlib.util.find_spec(config["module_path"])
+        # Try to import the module from various locations, return the first
+        # successful import, or None if they all failed
+        #
+        # 1. try to import the module directly off PYTHONPATH
+        # 2. try to import a module with the given name in the module_path
+        # 3. try to import the module_path itself
+        module_spec = None
+        namespaces = [
+            config["module"],
+            config["module_path"] + '.' + config["name"],
+            config["module_path"],
+        ]
+        for namespace in namespaces:
+            try:
+                module_spec = importlib.util.find_spec(namespace)
+                if module_spec:
+                    break
+            except (ModuleNotFoundError, AttributeError):
```

Comment:
The linter seems to be complaining that `ModuleNotFoundError` is not defined.

Reply:
Whoops, yea, missed this.  `ModuleNotFound` is a subclass of `ImportError` (https://airbrake.io/blog/python-exception-handling/class-hierarchy) so just changed it.